CONTRIBUTING.md

```diff
@@ -138,6 +138,52 @@ long as there have been changes.
 The current version number is automatically inferred from the last release (i.e.
 git tag), subsequent unreleased commits, and local changes, if any.
 
+## Nox
+
+`GLASS` supports running various critical commands using
+[nox](https://github.com/wntrblm/nox) to make them less intimidating for new
+developers. All of these commands (or sessions in the language of `nox`) -
+`lint`, `tests`, `coverage`, `doctests`, `docs`, and `build` - are defined in
+[noxfile.py](https://github.com/glass-dev/glass/main/noxfile.py).
+
+`nox` can be installed via `pip` using -
+
+```bash
+pip install nox
+```
+
+The default sessions (`lint` and `tests`) can be executed using -
+
+```bash
+nox
+```
+
+A particular session (for example `tests`) can be run with `nox` on all
+supported Python versions using -
+
+```bash
+nox -s tests
+```
+
+Only `tests`, `coverage`, and the `doctests` session run on all supported Python
+versions by default.
+
+To specify a particular Python version (for example `3.11`), use the following
+syntax -
+
+```bash
+nox -s tests-3.11
```

Comment:
Presumably, it is trivial to test on different OS with `nox` too?

Reply:
That should be just a matter of adding `if-else` within the sessions. For example, the tests session could be -
```py
@nox.session(python=ALL_PYTHON)
 def tests(session: nox.Session) -> None:
    """Run the unit tests."""
    session.install("-c", ".github/test-constraints.txt", "-e", ".[test]")
    if sys.platform != "win32":
        session.install("library not supported by windows")  # pytest should handle rest (`importorskip`, `skipif`, ...)
    session.run(
       "pytest",
       *session.posargs,
    )

Reply:
Okay. Am intrigued why you prefer `nox` over `tox` 😉 but we can talk about this offline